joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/CGlobal.scala

```diff
@@ -0,0 +1,43 @@
+package io.joern.c2cpg.astcreation
+
+import io.joern.x2cpg.datastructures.Global
+import java.util.concurrent.ConcurrentHashMap
+
+object CGlobal {
+
+  final case class MethodInfo(
+    name: String,
+    code: String,
+    fileName: String,
+    returnType: String,
+    astParentType: String,
+    astParentFullName: String,
+    lineNumber: Option[Int],
+    columnNumber: Option[Int],
+    lineNumberEnd: Option[Int],
+    columnNumberEnd: Option[Int],
+    signature: String,
+    offset: Option[(Int, Int)],
+    parameter: Seq[ParameterInfo],
+    modifier: Seq[String]
+  )
+  final class ParameterInfo(
```

Comment:
Is there a reason why MethodInfo is a case class and ParameterInfo is not?

Reply:
ParameterInfo has two mutable fields (to set var arg info). Scala case classes should never be mutable as it breaks some contracts when accessing them in a parallel environment (says the Scala doc).

Reply:
Missed the mutability part. Makes sense.